frontend/public/components/auto-complete.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import * as _ from 'lodash';
+import { useHideComponent } from '@console/shared';
+import { fuzzyCaseInsensitive } from './factory/table-filters';
+
+const MAX_SUGGESTIONS = 5;
+
+const labelParser = (nodes) => {
+  const labels = _.map(nodes, (x) => x.metadata.labels);
+  const stringify = (labelObj) =>
+    JSON.stringify(labelObj)
+      .replace(/[{}"]/g, '')
+      .split(',');
+  const flatLabels = _.flatten(_.map(labels, stringify));
+  const set = new Set(flatLabels);
+  return Array.from(set);
```

Comment:
Line 11-17 is used at 2 places. Better to create a common function and use it at both places.